src/main/java/com/arpnetworking/metrics/mad/model/DefaultRecord.java

```diff
@@ -198,6 +216,8 @@ protected void reset() {
         @NotNull
         private ZonedDateTime _time;
         @NotNull
```

Comment:
This field should be `@Nullable` right since it's not on all requests?

Reply:
Why @Nullable instead of `Optional<ZonedDateTime>`?

Reply:
Builder setters are used to deserialize from formats like JSON and HOCON and this must accept `null`. It was easier to have these setters always accept objects (e.g. `Double` not `double`) and deal with `null` on validation then to use `Optional` and get the parsers to do the conversion correctly in all cases because we definitely don't want to deal with an `Optional` that is either set, unset or `null`.